server/src/configuration_set.rs

```diff
@@ -68,18 +72,25 @@ impl ConfigurationSet {
         let include_built_in = parse_include_built_in(value);
 
         // Create the SliceConfig and CompilationState
-        let mut slice_config = SliceConfig::default();
-        slice_config.set_workspace_root_path(root_path.to_owned());
-        slice_config.set_built_in_slice_path(include_built_in.then(|| built_in_path.to_owned()));
-        slice_config.set_search_paths(paths);
-
+        let slice_config = SliceConfig {
+            paths,
+            workspace_root_path: Some(root_path.to_owned()),
+            built_in_slice_path: include_built_in.then(|| built_in_path.to_owned()),
+        };
         let compilation_data = CompilationData::default();
-        Self { slice_config, compilation_data }
+        Self { slice_config, compilation_data, cached_slice_options: None }
     }
 
     pub fn trigger_compilation(&mut self) -> Vec<Diagnostic> {
+        // Re-compute the `slice_options` we're going to pass into the compiler, if necessary.
```

Comment:
Do we ever actually "re-compute" this? it reads. more like it's just lazily initialized

Reply:
It's kind of both.

If the user changes their configuration, we need to re-compute it.
But we also do lazily re-compute it (we wait until we're about to compile).